custom_components/ramses_cc/remote.py

```diff
@@ -37,23 +70,30 @@ async def async_setup_platform(
 ) -> None:
     """Create remotes for HVAC."""
 
-    if discovery_info is None:  # or not discovery_info.get("remotes")  # not needed
-        return
+    broker: RamsesBroker = hass.data[DOMAIN][BROKER]
+
+    if not broker._services.get(PLATFORM):
+        broker._services[PLATFORM] = True
 
-    broker = hass.data[DOMAIN][BROKER]
+        platform = entity_platform.async_get_current_platform()
 
-    platform = entity_platform.async_get_current_platform()
+        platform.async_register_entity_service(
+            SVC_LEARN_COMMAND, SVC_LEARN_COMMAND_SCHEMA, "async_learn_command"
+        )
+        platform.async_register_entity_service(
+            SVC_SEND_COMMAND, SVC_SEND_COMMAND_SCHEMA, "async_send_command"
+        )
+        platform.async_register_entity_service(
+            SVC_DELETE_COMMAND, SVC_DELETE_COMMAND_SCHEMA, "async_delete_command"
+        )
 
-    for name, schema in SVCS_REMOTE.items():
-        platform.async_register_entity_service(name, schema, f"svc_{name}")
+    if discovery_info is None:
+        return
```

Comment:
Shouldn't this (`discovery_info`) be at the top, so the services are not registered, unless a remote exists?

Reply:
I believe the check is redundant as the platform setup only gets called by the broker if there is discovery info available. But I thought I'd leave complete removal of it until I PR changes relating to entity addition.

Reply:
IIRC, it has something to do with config flow too.

Reply:
For config flow, the entry gets forwarded to each platform. If we want to maintain only creating services if we have supported devices (which would be nicer) then the approach taken in the core zwave JS component will probably work quite nicely.

Reply:
> I believe the check is redundant as the platform setup only gets called by the broker if there is discovery info available. 

IIRC, when I submitted a PR to HA (I wrote the **evohome** integration, among others), a core dev had insisted I put it back... But, TBH, I can't be sure.

It will cost us nothing to leave it in, please (I am willing to be talked around, if you can e.g. show me an existing integration that is missing it).

I note new entity detection in in the broker/co-ordinator. I often wondered if it would be better to do that in the platform... probably not.

> If we want to maintain only creating services if we have supported devices (which would be nicer)

I agree - we *should not* create a service unless there is an entity for it to run against. That means, that some service for a platform will be registered, and other not.

> then the approach taken in the core zwave JS component will probably work quite nicely.

I have no capacity to research that atm. I would be pleased to accept a PR from you.

Reply:
I'll just shift the check back to the start for this PR; it will be more relevant to talk about it in the context of later changes and as you say does no harm for now.

Reply:
> IIRC, when I submitted a PR to HA (I wrote the **evohome** integration, among others), a core dev had insisted I put it back... But, TBH, I can't be sure.

For reference: https://github.com/home-assistant/core/pull/26042#discussion_r319737004 which I suppose makes sense whilst we're still using discovery.

Reply:
> For reference: https://github.com/home-assistant/core/pull/26042#discussion_r319737004 which I suppose...

You found it!